File: 2023/C/estoque.c
#include <stdio.h>

int main(void) {
    int tipos, tamanhos;
    scanf("%d %d", &tipos, &tamanhos);

    char estoque[tipos][tamanhos];
    for (int tipo = 0; tipo < tipos; tipo++) {
        for(int tamanho = 0; tamanho < tamanhos; tamanho++) {
            scanf("%d", &estoque[tipo][tamanho]);
        }
    }

    int pedidos;
    scanf("%d", &pedidos);

    int pedidos_efetivados = 0;
    for (int pedido = 0; pedido < pedidos; pedido++) {
        int tipo, tamanho;
        scanf("%d %d", &tipo, &tamanho);
        tipo--;
        tamanho--;

        if (estoque[tipo][tamanho] > 0) {
            estoque[tipo][tamanho]--;
            pedidos_efetivados++;
        }
    }

    printf("%d\n", pedidos_efetivados);
    return 0;
}

File: 2023/python/estoque.py
tipos, tamanhos = [int(n) for n in input().split(" ")]

estoque = []
for i in range(tipos):
    estoque.append([int(quantidade) for quantidade in input().split(" ")])

pedidos_efetivados = 0
for i in range(int(input())):
    tipo, tamanho = [int(n) - 1 for n in input().split()]
    if estoque[tipo][tamanho] == 0:
        continue

    estoque[tipo][tamanho] -= 1
    pedidos_efetivados += 1

print(pedidos_efetivados)